docs/External-Commands.md

```diff
@@ -66,3 +68,87 @@ Note they are largely untested, and as always, be careful about running untested
 Install any `gem` package into a self-contained Homebrew Cellar location: <https://github.com/sportngin/brew-gem>
 
 Note this can also be installed with `brew install brew-gem`.
+
+## External commands in taps
+External commands can be hosted in a [tap](Taps.md) to allow users to easy install and use the command. See [How to Create and Maintain a Tap](How-to-Create-and-Maintain-a-Tap.md) for more details about creating and maintaining a tap.
+
+External commands should be added to a `cmd` directory in the tap. An external command `extcmd` implemented as a Ruby command should live in `cmd/extcmd.rb` (don't forget to `chmod +x`).
+
+To easily use Homebrew's argument parser, follow the following Ruby template for external commands (replacing all instances of `foo` with the name of the command):
+
+```ruby
+# frozen_string_literal: true
+
+module Homebrew
+  module_function
+
+  def foo_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `foo` [<options>] <formula|cask>
```

Comment:
Not blocking this PR but: this is the part I'm wondering if we could just generate based on the `named_args` work you've done? In a bunch of cases it ends up being incorrect/outdated.

Reply:
Yeah, I'd like to be able to do that. It is on my list of things to do with the `named_args` stuff. I'm prioritizing the completions, though, so I was planning on doing this later. Let me know if I should reprioritize.

Reply:
Actually, this is probably much simpler than I thought. I definitely want to get the bash completions done soon but maybe I'll take a quick break for this... I'll think about it